libarchive/write.py

```diff
@@ -73,6 +74,39 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry):
```

Comment:
I like more consistency of one object with arguments than 3 arguments but I'm open to discussion and I can change that if you want.